addons/portal/static/src/js/portal.js

```diff
@@ -75,22 +75,42 @@ publicWidget.registry.PortalHomeCounters = publicWidget.Widget.extend({
      */
     async _updateCounters(elem) {
         const numberRpc = 3;
-        const needed = this.$('[data-placeholder_count]')
-                                .map((i, o) => $(o).data('placeholder_count'))
-                                .toArray();
+        const needed = Object.values(this.el.querySelectorAll('[data-placeholder_count]'))
+                                .map(o => o.dataset['placeholder_count']);
         const counterByRpc = Math.ceil(needed.length / numberRpc);  // max counter, last can be less
+        const countersToKeep = this.getCountersToKeep();
 
         const proms = [...Array(Math.min(numberRpc, needed.length)).keys()].map(async i => {
-            await this._rpc({
+            const data = await this._rpc({
                 route: "/my/counters",
                 params: {
                     counters: needed.slice(i * counterByRpc, (i + 1) * counterByRpc)
                 },
-            }).then(data => {
-                Object.keys(data).map(k => this.$("[data-placeholder_count='" + k + "']").text(data[k]));
             });
+            Object.keys(data).forEach(counter => {
+                const portalLineCounter = this.el.querySelector(`[data-placeholder_count='${counter}']`);
+                portalLineCounter.textContent = data[counter];
+                if (data[counter] !== 0 || countersToKeep.includes(counter)) {
+                    portalLineCounter.parentElement.classList.remove('d-none');
+                }
+            });
+            return data;
+        });
+        return Promise.all(proms).then((results) => {
+            const counterValues = Object.values({...results[0], ...results[1], ...results[2]});
```

Comment:
From what I understand there are not always exactly 3 rpcs?

Would be better to take a look at something like:
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

Seems it would be the right tool for this.